### ISSUE TITLE
This is a problem that occurs in the Object Detection Model.

### ISSUE THREAD

**Issue by hagsac** [stereolabs/zed-unity#179]
The default setting is
MULTI_CLASS_BOX_ACCURATE.
I have confirmed normal operation.
However, if I change the setting to MULTI_CLASS_BOX or any other model, the program stops as soon as it runs in Unity.

MULTI_CLASS_BOX_ACCURATE : O
MULTI_CLASS_BOX : X
MULTI_CLASS_BOX_MEDIUM : X
HUMAN_BODY_FAST : X
HUMAN_BODY_MEDIUM :  O
HUMAN_BODY_ACCURATE  : X

windows 10
unity  2020.3.6f1 personal
CUDA Toolkit 11.1
zed sdk 3.5  cuda 11.1
zed-Unity v3.5.0

**Comment by Bvallon-sl** [comment]
Hi,

Thanks for reaching us out.

Does Unity crash ? If so, can you share your crash report ?
If Unity is "just" freezing, it's normal as the first time you use any Detection Model, it downloads, and optimizes it. It can take 20 or 30sec to start, and after that it will be a lot faster.

Best regards,
Benjamin Vallon

**Comment by hagsac** [comment]
Thank you always for your kind reply.
I am not a citizen of an English-speaking country, so there will be many problems with English sentences.


It's settled now. The reason for that is not accurate.
However, after several attempts, we reinstalled cuda, SDK, and Unity assets.
cuda 11.1
unity 2021
sdk 3.5
unity plugin 3.5

1. It is currently operating normally and cannot be copied, but [DonDestroyLoad] has been created in the diary to prevent shutdown.
At the time of the trouble,
When the model failed to function, [DonDestroyLoad] was not created and [unity] was completely interrupted and automatically terminated.


2. We found another problem.
When I play images with [vehicle] and [person], the model works accurately for a certain time.
However, the moment [vehicle] is detected, [unity] stops and ends completely.
Both my [SVO] file and Sterolabs' [ZED2_HD1080_SN24605000_Street_H264.svo] sample file have the same phenomenon.
The file above always stops at around 765 frames.



[에러]
---------------------------------------------------------------------------------------------------------------------------
ransform.position assign attempt for 'Object #34' is not valid. Input position is { NaN, 1.585469, NaN }.
UnityEngine.Transform:set_position (UnityEngine.Vector3)
ZED3DObjectVisualizer:Visualize3DBoundingBoxes (DetectionFrame) (at Assets/ZED/Examples/Object Detection/Scripts/ZED3DObjectVisualizer.cs:189)
ZEDManager:UpdateObjectsDetection () (at Assets/ZED/SDK/Helpers/Scripts/ZEDManager.cs:2811)
ZEDManager:Update () (at Assets/ZED/SDK/Helpers/Scripts/ZEDManager.cs:2498)
-------------------------------------------------------------------------------------------------------------------------------
...
...
-------------------------------------------------------------------------------------------------------------------------------
transform.localScale assign attempt for 'ID Canvas' is not valid. Input localScale is { Infinity, 1.401994, Infinity }.
UnityEngine.Transform:set_localScale (UnityEngine.Vector3)
BBox3DHandler:OnCameraPreRender (UnityEngine.Camera) (at Assets/ZED/Examples/Object Detection/Scripts/BBox3DHandler.cs:334)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
